### PR TITLE
Update github action workflows and upgrade ckan

### PR DESCRIPTION
**Pull request by kentsanggds**
## What

Upgrade of CKAN to 2.9.9 to address some security issues.
Update of the github actions workflows 
  - enable the creation of images based on branches so that they can be deployed onto the cluster
  - add ability to create the base images for the CI and local dev manually so that the run time on github actions will be reduced and dev cycle improved.

## Reference 

https://trello.com/c/bEsbEk1m/1151-run-security-check-and-resolve-critical-issues

### CODE REVIEW

File: docker/ckan/2.9.9-core.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# See CKAN docs on installation from Docker Compose on usage
```
Comment (sengi) [inline docker/ckan/2.9.9-core.Dockerfile:1]: optional: I'd recommend running [Hadolint](https://github.com/hadolint/hadolint#haskell-dockerfile-linter) on this file and considering its suggestions. (It occasionally comes up with false positives but mostly its suggestions are good.)

**Review by sengi — APPROVED** [review]
I've not much more than skimmed this because there are a lot of commits here, but I didn't spot anything bad and I'm conscious you've already been waiting on a review for a while so 🚀

File: docker/build-image.sh [hunk 2]
```diff
@@ -2,16 +2,36 @@
 
 set -eux
 
-DOCKER_TAG="${GITHUB_SHA}"
+build () {
+  if [ "${ARCH}" = "amd64" ]; then
+    docker build . -t ghcr.io/alphagov/${APP}:${1} -f docker/${APP}/${1}.Dockerfile
```
[the remark this answers is not in the document]
Comment (kentsanggds) [reply docker/build-image.sh:7]: ok, I'll run it through that

File: docker/ckan/2.9.7.Dockerfile [hunk 3]
```diff
@@ -101,15 +101,20 @@ ENV ckan_harvest_sha='cb0a7034410f217b2274585cb61783582832c8d5'
...
 ENV ckan_spatial_fork='alphagov'
 
 RUN echo "pip install DGU extensions..." && \
 
     pip install $pipopt -U $(curl -s https://raw.githubusercontent.com/$ckan_dcat_fork/ckanext-dcat/$ckan_dcat_sha/requirements.txt) && \
     pip install $pipopt -U "git+https://github.com/$ckan_dcat_fork/ckanext-dcat.git@$ckan_dcat_sha#egg=ckanext-dcat" && \
 
-    pip install $pipopt -U $(curl -s https://raw.githubusercontent.com/$ckan_spatial_fork/ckanext-spatial/$ckan_spatial_sha/requirements.txt) && \
+    # pip install $pipopt -U $(curl -s https://raw.githubusercontent.com/$ckan_spatial_fork/ckanext-spatial/$ckan_spatial_sha/requirements.txt) && \
+    # pip install $pipopt -U "git+https://github.com/$ckan_spatial_fork/ckanext-spatial.git@$ckan_spatial_sha#egg=ckanext-spatial" && \
```
[the remark this answers is not in the document]
Comment (kentsanggds) [reply docker/ckan/2.9.7.Dockerfile:113]: yes, will remove, they were added in by accident

File: docker/ckan/2.9.9-core.Dockerfile (hunk 1 above)
Comment (kentsanggds) [reply docker/ckan/2.9.9-core.Dockerfile:1]: ok, will try it out

**Pull request merged**